v4/newrelic/transaction.go

```diff
@@ -260,6 +304,7 @@ func (txn *Transaction) NewGoroutine() *Transaction {
 	newTxn := *txn
```

Comment:
This doesn't need to be done under lock does it?

Reply:
You're right, it should be under a lock. We're copying information from `txn` in a non-atomic way.